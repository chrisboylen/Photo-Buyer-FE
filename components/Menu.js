import React, { Component } from "react";
import Link from "next/link";
import { MobileStyles } from "./styles/NavStyles";
import User from "../components/User";
import Signout from "./Signout";
import { Mutation } from "react-apollo";
import { TOGGLE_CART_MUTATION } from "./Cart";
import CartCount from "./CartCount";
import PropTypes from "prop-types";

class Menu extends Component {
  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <MobileStyles data-test="mobile">
            <Link href="/items">
              <a data-test="items" onClick={this.props.toggleMenu}>
                Shop
              </a>
            </Link>
            {me && (
              <>
                <Link href="/sell">
                  <a onClick={this.props.toggleMenu}>Sell</a>
                </Link>
                <Link href="/orders">
                  <a onClick={this.props.toggleMenu}>Orders</a>
                </Link>
                <Signout />
                <Mutation mutation={TOGGLE_CART_MUTATION}>
                  {toggleCart => (
                    <button onClick={toggleCart} data-test="signout">
                      My Cart
                      <CartCount
                        count={me.cart.reduce(
                          (tally, cartItem) => tally + cartItem.quantity,
                          0
                        )}
                      />
                    </button>
                  )}
                </Mutation>
              </>
            )}
            {!me && (
              <>
                <Link href="/signup">
                  <a onClick={this.props.toggleMenu}>Signup</a>
                </Link>
                <Link href="/signin">
                  <a onClick={this.props.toggleMenu}>Signin</a>
                </Link>
              </>
            )}
          </MobileStyles>
        )}
      </User>
    );
  }
}

Menu.propTypes = {};

export default Menu;
