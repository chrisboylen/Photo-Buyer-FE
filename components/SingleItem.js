import React, { Component } from "react";
import gql from "graphql-tag";
import { Query } from "react-apollo";
import Error from "./ErrorMessage";
import styled from "styled-components";
import Head from "next/head";
import formatMoney from "../lib/formatMoney";

const SingleItemStyles = styled.div`
  border-radius: 5px;
  max-width: 1200px;
  margin: 1rem auto;
  box-shadow: ${props => props.theme.bs};
  display: grid;
  grid-auto-columns: 1fr;
  grid-auto-flow: column;
  min-height: 800px;
  h2 {
    background: ${props => props.theme.blue};
    border-radius: 5px 5px 0 0;
    color: ${props => props.theme.white};
    text-align: center;
    margin: 0;
    span {
      color: #c7e4ff;
    }
  }
  img {
    width: 100%;
    height: 100%;
    object-fit: contain;
    border-radius: 0 0 5px 5px;
  }
  .details {
    margin: 3rem;
    font-size: 2rem;
  }
`;

const SINGLE_ITEM_QUERY = gql`
  query SINGLE_ITEM_QUERY($id: ID!) {
    item(where: { id: $id }) {
      id
      title
      description
      largeImage
    }
  }
`;
class SingleItem extends Component {
  render() {
    return (
      <Query
        query={SINGLE_ITEM_QUERY}
        variables={{
          id: this.props.id
        }}
      >
        {({ error, loading, data }) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading...</p>;
          if (!data.item) return <p>No Item Found for {this.props.id}</p>;
          const item = data.item;
          return (
            <SingleItemStyles>
              <Head>
                <title>Photo Buyer | {item.title}</title>
              </Head>
              <div className="details">
                <h2>
                  <span>Viewing: </span> {item.title}
                </h2>
                <img src={item.largeImage} alt={item.title} />
                <p>{item.description}</p>
              </div>
            </SingleItemStyles>
          );
        }}
      </Query>
    );
  }
}

export default SingleItem;
export { SINGLE_ITEM_QUERY };
