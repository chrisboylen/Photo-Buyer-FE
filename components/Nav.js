import React, { Component } from "react";
import Link from "next/link";
import { NavStyles, MobileStyles } from "./styles/NavStyles";
import User from "./User";
import Signout from "./Signout";
import { Mutation } from "react-apollo";
import { TOGGLE_CART_MUTATION } from "./Cart";
import CartCount from "./CartCount";
import Menu from "./Menu";

class Nav extends Component {
  state = {
    isOpen: false
  };

  toggleMenu = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <NavStyles data-test="nav">
            {this.state.isOpen && (
              <Menu isOpen={this.state.isOpen} toggleMenu={this.toggleMenu} />
            )}
            <button
              onClick={this.toggleMenu}
              className="mobile-checkbox"
              id="mobile-toggle"
            />
            <label htmlFor="mobile-toggle" className="mobile-btn">
              <span
                className={
                  this.state.isOpen
                    ? "mobile-icon mobile-icon-open"
                    : "mobile-icon"
                }
              >
                &nbsp;
              </span>
            </label>
            <div className="mobile-bkg">&nbsp;</div>
            <div className="web-nav">
              <Link href="/items">
                <a>Shop</a>
              </Link>
              {me && (
                <>
                  <Link href="/sell">
                    <a>Sell</a>
                  </Link>
                  <Link href="/orders">
                    <a>Orders</a>
                  </Link>
                  <Signout />
                  <Mutation mutation={TOGGLE_CART_MUTATION}>
                    {toggleCart => (
                      <button onClick={toggleCart}>
                        My Cart
                        <CartCount
                          count={me.cart.reduce(
                            (tally, cartItem) => tally + cartItem.quantity,
                            0
                          )}
                        />
                      </button>
                    )}
                  </Mutation>
                </>
              )}
              {!me && (
                <>
                  <Link href="/signup">
                    <a>Signup</a>
                  </Link>
                  <Link href="/signin">
                    <a>Signin</a>
                  </Link>
                </>
              )}
            </div>
          </NavStyles>
        )}
      </User>
    );
  }
}

export default Nav;
