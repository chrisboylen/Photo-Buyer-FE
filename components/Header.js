import Link from "next/link";
import styled from "styled-components";
import Nav from "./Nav";
import Router from "next/router";
import NProgress from "nprogress";
import Cart from "./Cart";
import Search from "./Search";

Router.onRouteChangeStart = () => {
  NProgress.start();
};
Router.onRouteChangeComplete = () => {
  NProgress.done();
};
Router.onRouteChangeError = () => {
  NProgress.done();
};

const Logo = styled.h1`
  background: ${props => props.theme.lightblue};
  font-size: 8rem;
  margin: 0;
  min-height: 12rem;
  position: relative;
  z-index: 1500;
  a {
    margin: 0 0 0 10rem;
    padding: 1rem 2rem 0.6rem 2rem;
    border-radius: 8px;
    background: #e8f4ff;
    color: #4dadfe;
    text-transform: uppercase;
    text-decoration: none;
  }
  span {
    background: linear-gradient(to right, #00f1fe 0%, #4dadfe 100%);
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
  img {
    width: 6rem;
    margin: 0 2rem 0 0;
  }
  @media (max-width: 1300px) {
    font-size: 6rem;
    img {
      width: 4.5rem;
    }
    @media (max-width: 760px) {
      line-height: 1.5;
      a {
        top: 2rem;
        left: 3rem;
        margin: 0;
        padding: 0.3rem 1.1rem 0 1.1rem;
        position: relative;
      }
      img {
        margin: 0;
        width: 4rem;
      }
      span {
        display: none;
      }
    }
  }
`;

const StyledHeader = styled.header`
  left: 0;
  top: 0;
  position: sticky;
  z-index: 500;
  .bar {
    grid-template-columns: auto 1fr;
    justify-content: space-between;
    align-items: stretch;
    @media (max-width: 1300px) {
      grid-template-columns: 1fr;
      justify-content: center;
    }
  }
  .sub-bar {
    background: linear-gradient(
      to bottom,
      #8cbce2 0%,
      #8cbce2 70%,
      #193e5d 100%
    );
    display: grid;
    grid-template-columns: 1fr;
    border-bottom: 1px solid ${props => props.theme.lightgrey};
  }
`;

const Header = () => (
  <StyledHeader>
    <div className="bar">
      <Logo>
        <Link href="/">
          <a>
            <img src="../static/camera.png" alt="Camera Icon" />
            <span className="text">Photo Buyer</span>
          </a>
        </Link>
      </Logo>
      <Nav />
    </div>
    <div className="sub-bar">
      <Search />
    </div>
    <Cart />
  </StyledHeader>
);

export default Header;
