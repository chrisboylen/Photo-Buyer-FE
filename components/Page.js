import React, { Component } from "react";
import styled, { ThemeProvider, injectGlobal } from "styled-components";
import Header from "./Header";
import Meta from "./Meta";

const theme = {
  lightblue: "#9ed2ff",
  blue: "#4dadfe",
  black: "#193e5d",
  grey: "#3a3a3a",
  lightgrey: "#b1c0e0",
  white: "#e8f4ff",
  offWhite: "#ededed",
  maxWidth: "1000px",
  bs: "0 12px 24px 0 rgba(0, 0, 0, 0.21)"
};

const StyledPage = styled.div`
  color: ${props => props.theme.black};
`;

const Inner = styled.div`
  max-width: ${props => props.theme.maxWidth};
  margin: 0 auto;
  padding: 2rem;
  @media (max-width: 760px) {
    padding: 0;
  }
`;

injectGlobal`
  @font-face {
    font-family: 'DoHyeon-Regular';
    src: url('/static/DoHyeon-Regular.ttf')
    format('woff2');
    font-weight: normal;
    font-style: normal;
  }
  html {
    box-sizing: border-box;
    font-size: 10px;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  body {
    background: #e8f4ff52;
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: 'DoHyeon-Regular';
  }
  a {
    text-decoration: none;
    color: ${theme.black}
  }
`;

class Page extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <StyledPage>
          <Meta />
          <Header />
          <Inner>{this.props.children}</Inner>
        </StyledPage>
      </ThemeProvider>
    );
  }
}

export default Page;
