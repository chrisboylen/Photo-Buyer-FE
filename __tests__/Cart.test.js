import Cart, { LOCAL_STATE_QUERY } from "../components/Cart";
import { CURRENT_USER_QUERY } from "../components/User";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import { fakeUser, fakeCartItem, fakeCartItems } from "../lib/testUtils";

const mocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me: {
          ...fakeUser(),
          cart: [fakeCartItem()]
        }
      }
    }
  },
  {
    request: { query: LOCAL_STATE_QUERY },
    result: { data: { cartOpen: true } }
  }
];
const mocks2 = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me: {
          ...fakeUser(),
          cart: [fakeCartItems()]
        }
      }
    }
  },
  {
    request: { query: LOCAL_STATE_QUERY },
    result: { data: { cartOpen: true } }
  }
];

describe("CART", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Cart />
      </MockedProvider>
    );
  });

  it("renders and matches snapshot with cart item", async () => {
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find("header"))).toMatchSnapshot();
    expect(wrapper.find("CartItem")).toHaveLength(1);
  });

  it("renders and matches snapshot with cart items", async () => {
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find("header"))).toMatchSnapshot();
  });
});
