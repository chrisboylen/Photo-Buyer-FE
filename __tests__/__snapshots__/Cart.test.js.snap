// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CART renders and matches snapshot 1`] = `
<header>
  <CloseButton
    onClick={[Function]}
    title="close"
  >
    <button
      className="CloseButton-sc-7151hm-0 kktGeb"
      onClick={[Function]}
      title="close"
    >
      Ã—
    </button>
  </CloseButton>
  <Supreme>
    <h3
      className="Supreme-xza65r-0 hMFNtA"
    >
      Miss Coleman Berge
      's Cart
    </h3>
  </Supreme>
  <p>
    You have 
    1
     Item
     in your cart.
  </p>
</header>
`;
